### PR TITLE
Add steps to take eck dump

### PR DESCRIPTION
**Pull request by kunisen**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? 

Yes.

- Have you followed the [contributor guidelines](https://github.com/elastic/cloud-on-k8s/tree/master/CONTRIBUTING.md)? 
 
Yes

- If you submit code, is your pull request against master? We recommend pull requests against master. We will backport them as needed. 

This is not a code submission but a doc PR.

### CODE REVIEW

**Comment by thbkrkr** [comment]
jenkins test this please

**Review by alaudazzi — APPROVED** [review]
LGTM, with a few minor editing suggestions.

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+== Prepare
+
+To download the ECK dump shell file, run:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+curl -L -O https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/support/diagnostics/eck-dump.sh
+------------------------------------------------
+
+Change mode to executable.
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:23]:
```suggestion
Change mode to executable:
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Steps to take ECK dump:
+
+[float]
+== Prepare
+
+To download the ECK dump shell file, run:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+curl -L -O https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/support/diagnostics/eck-dump.sh
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:20]: Nit: I would combine the two commands for simplicity:
```
curl -L -O https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/support/diagnostics/eck-dump.sh && chmod +x eck-dump.sh
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+[source,bash]
+----
+./eck-dump.sh -o <path/to/output/dump/files> [-z] [-v] -N <operator namespace> -n <resources namespace>
+----
+
+
+[float]
+== Example
+
+Here is an example.
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:56]:
```suggestion
Here is an example:
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+- Assuming operator & resources are both in `elastic-system` namespace
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:58]: I would prefer to use a different namespace than the one used to deploy the operator, to deploy the resources.
And even why not two namespaces:
```
-N "elastic-system" -n "security,monitoring"
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+[source,bash]
+----
+./eck-dump.sh -o eck-dump-log -z -N "elastic-system" -n "elastic-system"
+----
+
+Sample output
+
+[source,bash]
+----
+ECK info dumped to eck-dump-log-15_ 7_2021_15_52_45.tgz
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:71]: Did you really get a white space?
```suggestion
ECK info dumped to eck-dump-log-15_07_2021_15_52_45.tgz
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+----
+
+Directory structure of the output files. The output may vary depending on the configuration
+
+[source,bash]
+----
+% tree -A -L 10      
+
+.
+├── clusterroles.txt
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:81]:
```suggestion
> tree -A -L 10 eck-dump-log                                              
eck-dump-log
├── clusterroles.txt
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc (hunk 2 above)
Comment (kunisen) [reply docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:20]: Thanks for the comment here! For simplicity, I am very happy to do it as you advised!
But sorry I am not very sure how to delete the "Change mode to executable" section in my later part.
May I trouble you to help with removing the corresponding statement please?

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc (hunk 5 above)
Comment (kunisen) [reply docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:71]: Unfortunately, there's that space... I tested it again (I am pretty sure we can ignore the error message as I didn't add APM component when I did the test again.
```
./eck-dump.sh -o eck-dump-log -z -N "elastic-system" -n "elastic-system"
error: the server doesn't have a resource type "elasticmapsserver"
ECK info dumped to eck-dump-log-11_ 8_2021_19_41_02.tgz
```

**Review by thbkrkr — APPROVED** [review]
LGTM

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+- Assuming the operator namespace is `elastic-system` and resources namespaces are `security` and `monitoring`
+- Generate dump log to current folder with `eck-dump-log` prefix
+- Also zip it
+
+[source,bash]
+----
+./eck-dump.sh -o eck-dump-log -z -N "elastic-system" -n "security,monitoring"
+----
+
+Sample output
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:60]:
```suggestion
Sample output:
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+│   ├── pods.json
+│   ├── replicasets.json
+│   ├── secrets.json
+│   ├── services.json
+│   └── statefulsets.json
+├── nodes.json
+├── podsecuritypolicies.json
+└── version.json
+
+9 directories, 33 files
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:116]: Nit: dump with the two namespaces used in the sample command above.
```suggestion
 tree -A -L 10 eck-dump-log      
eck-dump-log
├── clusterroles.txt
├── elastic-system
│   ├── agent.json
│   ├── apmserver.json
│   ├── beat.json
│   ├── configmaps.json
│   ├── controllerrevisions.json
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── elasticmapsserver.json
│   ├── elasticsearch.json
│   ├── endpoints.json
│   ├── enterprisesearch.json
│   ├── events.json
│   ├── kibana.json
│   ├── networkpolicies.json
│   ├── persistentvolumeclaims.json
│   ├── persistentvolumes.json
│   ├── pod
│   │   └── elastic-operator-0
│   │       └── logs.txt
│   ├── poddisruptionbudgets.txt
│   ├── pods.json
│   ├── replicasets.json
│   ├── secrets.txt
│   ├── services.json
│   └── statefulsets.json
├── monitoring
│   ├── agent.json
│   ├── apmserver.json
│   ├── beat.json
│   ├── configmaps.json
│   ├── controllerrevisions.json
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── elasticmapsserver.json
│   ├── elasticsearch.json
│   ├── endpoints.json
│   ├── enterprisesearch.json
│   ├── events.json
│   ├── kibana.json
│   ├── networkpolicies.json
│   ├── persistentvolumeclaims.json
│   ├── persistentvolumes.json
│   ├── pod
│   │   ├── apm-01-prod-apm-server-8485b5bf97-j6g25
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-apm-server-8485b5bf97-jxc6g
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-es-master-0
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-kb-b56645bb9-4kxlq
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-kb-b56645bb9-5m2mb
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-kb-c6d58674f-bcd62
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-kb-c6d58674f-cjfzm
│   │   │   └── logs.txt
│   │   ├── apm-01-prod-kb-df76dbfff-9nhvx
│   │   │   └── logs.txt
│   │   ├── logs-01-prod-es-master-0
│   │   │   └── logs.txt
│   │   ├── logs-01-prod-es-master-1
│   │   │   └── logs.txt
│   │   ├── logs-01-prod-es-master-2
│   │   │   └── logs.txt
│   │   ├── logs-01-prod-kb-8599b565cf-4s9n2
│   │   │   └── logs.txt
│   │   ├── logs-01-prod-kb-8599b565cf-cx5wv
│   │   │   └── logs.txt
│   │   ├── logs-02-prod-es-master-0
│   │   │   └── logs.txt
│   │   ├── logs-02-prod-kb-7fff9c5fc6-pnlb2
│   │   │   └── logs.txt
│   │   ├── metrics-01-prod-es-master-0
│   │   │   └── logs.txt
│   │   └── metrics-01-prod-kb-7ffccd6747-cdqhw
│   │       └── logs.txt
│   ├── poddisruptionbudgets.txt
│   ├── pods.json
│   ├── replicasets.json
│   ├── secrets.txt
│   ├── services.json
│   └── statefulsets.json
├── nodes.json
├── podsecuritypolicies.json
├── security
│   ├── agent.json
│   ├── apmserver.json
│   ├── beat.json
│   ├── configmaps.json
│   ├── controllerrevisions.json
│   ├── daemonsets.json
│   ├── deployments.json
│   ├── elasticmapsserver.json
│   ├── elasticsearch.json
│   ├── endpoints.json
│   ├── enterprisesearch.json
│   ├── events.json
│   ├── kibana.json
│   ├── networkpolicies.json
│   ├── persistentvolumeclaims.json
│   ├── persistentvolumes.json
│   ├── pod
│   │   ├── siem-01-prod-es-master-0
│   │   │   └── logs.txt
│   │   ├── siem-01-prod-kb-554b496d8c-hk8js
│   │   │   └── logs.txt
│   │   ├── siem-02-prod-es-master-0
│   │   │   └── logs.txt
│   │   └── siem-02-prod-kb-855f7dbdc-bh6ks
│   │       └── logs.txt
│   ├── poddisruptionbudgets.txt
│   ├── pods.json
│   ├── replicasets.json
│   ├── secrets.txt
│   ├── services.json
│   └── statefulsets.json
└── version.json

28 directories, 92 files
```

File: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+----
+Options:
+  -N, --operator-namespaces     Namespace(s) in which operator(s) are running in (comma-separated list)
+  -n, --resources-namespaces    Namespace(s) in which resources are managed      (comma-separated list)
+  -o, --output-directory        Path to output dump files
+  -z, --create-zip              Create an archive with the dump files (implies --output-directory)
+  -v, --verbose                 Verbose mode
+----
+
+Command
```
Comment (thbkrkr) [inline docs/operating-eck/troubleshooting/take-eck-dump.asciidoc:38]:
```suggestion
Command:
```

**Review by thbkrkr — COMMENTED** [review]
Found a few last nits.

**Pull request merged**